frontend/app_flowy/packages/flowy_editor/lib/undo_manager.dart

```diff
@@ -0,0 +1,111 @@
+import 'dart:collection';
+
+import 'package:flowy_editor/document/selection.dart';
+import 'package:flowy_editor/operation/operation.dart';
+import 'package:flowy_editor/operation/transaction_builder.dart';
+import 'package:flowy_editor/operation/transaction.dart';
+import 'package:flowy_editor/editor_state.dart';
+
+class HistoryItem extends LinkedListEntry<HistoryItem> {
+  final List<Operation> operations = [];
+  Selection? beforeSelection;
+  Selection? afterSelection;
+  bool _sealed = false;
+
+  HistoryItem();
+
+  seal() {
+    _sealed = true;
+  }
+
+  add(Operation op) {
+    operations.add(op);
+  }
+
+  addAll(Iterable<Operation> iterable) {
+    operations.addAll(iterable);
+  }
+
+  bool get sealed {
```

Comment:
move this line to line 17 would be better? Because the related logic will be more compact